src_c/alphablit.c

```diff
@@ -2707,209 +2685,421 @@ alphablit_alpha_sse2_argb_surf_alpha_big_end (SDL_BlitInfo * info)
        SDL1, as follows:
        dstRGB = (((dstRGB << 8) + (srcRGB - dstRGB) * srcA + srcRGB) >> 8)
          dstA = srcA + dstA - ((srcA * dstA) / 255);
-                                                        */
+    */
 
-    while (height--)
+    /* There are three paths through this blitter:
+
+        1. Two pixels at once - 64 bit edition.
+        2. Two pixels at once - 32 bit edition.
+        3. One pixel at a time.
+    */
+
+    if (((width % 2) == 0) && ((srcskip % 2) == 0) && ((dstskip % 2) == 0))
     {
-        LOOP_UNROLLED4(
-        {
-            Uint32 src_alpha = (*srcp & src_amask);
-            Uint32 dst_alpha = (*dstp & dst_amask) + dst_opaque;
-            /* modulate src_alpha - need to do it here for
-               accurate testing */
-            src_alpha = (src_alpha * modulateA) / 255;
+        width = width/2;
+        srcskip = srcskip/2;
+        dstskip = dstskip/2;
 
-            if ((src_alpha == src_amask) || (dst_alpha == 0))
-            {
-                /* 255 src alpha or 0 dst alpha
-                   So copy src pixel over dst pixel, also copy
-                   modulated alpha */
-                *dstp = (*srcp & 0xFFFFFF00) | src_alpha;
-            }
-            else
+        Uint64          *srcp = (Uint64 *)info->s_pixels;
+        Uint64          *dstp = (Uint64 *)info->d_pixels;
+
+        Uint64          src_amask = ((Uint64)srcfmt->Amask << 32) | srcfmt->Amask;
+        Uint64          dst_amask = ((Uint64)dstfmt->Amask << 32) | dstfmt->Amask;
+
```

Comment:
You might have figured it out by the time you read this, but this is what the old MS VC 9.0 compiler freaks out about.

It mostly follows C90 which forced you to have all your variables declared at the top of the function (or at the top of a code block).

Reply:
Yeah I knew it was going almost certainly going to complain about some variable declarations :) Just thought it would be easier to have the CI tell me where to look while I wrote up some other info.